pkg/ansibleapp/registry.go

```diff
@@ -0,0 +1,28 @@
+package ansibleapp
+
+type RegistryConfig struct {
+	Name string
+	Url  string
+}
+
+type Registry interface {
+	Init(RegistryConfig) error
+	LoadApps() error
+}
+
+func CreateRegistry(config RegistryConfig) Registry {
```

Comment:
I'd like to do some kind of self-registration by concrete registries with the factory. Think it can be done with `func init`, but need to explore that more.